src/runtime/runtime_rp2350.go

```diff
@@ -14,3 +15,60 @@ const (
 	sioIrqFifoProc0 = rp.IRQ_SIO_IRQ_FIFO
 	sioIrqFifoProc1 = rp.IRQ_SIO_IRQ_FIFO
 )
+
+// Due to hardware errata RP2350-E2 spinlocks are emulated in software as in pico-sdk.
+type spinLock struct {
+	// state field must be first field so its address is the same as the struct address.
+	state  uint8
+	id     uint8
```

Comment:
Why is there an `id` field here? It doesn't seem to be used?

Reply:
I originally just carried it over, but it's actually used for the hardware spinlocks in runtime_rp2.go:
```
/root/project/src/runtime/runtime_rp2.go:294:27: unknown field id in struct literal of type spinLock
/root/project/src/runtime/runtime_rp2.go:295:27: unknown field id in struct literal of type spinLock
/root/project/src/runtime/runtime_rp2.go:296:27: unknown field id in struct literal of type spinLock
/root/project/src/runtime/runtime_rp2.go:297:27: unknown field id in struct literal of type spinLock
```